In [1]:
from train_classifieur import train_classifier, pred_classifier

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Au vue de la puissance de nos ordinateurs (1 epoch prend 20 minutes), nous avons décidé de limitter le classifier à 2 epochs.

Training avec les metadata originelles avec les weights entièrement à 1 :

In [2]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/metadata.csv",
    max_epochs = 2
)

GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs


Start training


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:654: Checkpoint directory C:\Users\firda\Documents\L3\fairness IA\final\Fairness\expe_log exists and is not empty.
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\optim\lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params
0         Non-trainable params
11.2 M    Total params
44.710    Total estimated model params size (MB)
69      

Sanity Checking: |          | 0/? [00:00<?, ?it/s]

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [03:29<00:00,  0.61it/s, v_num=0]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [03:30<00:00,  0.61it/s, v_num=0]
End of training 492.30383229255676


In [3]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/metadata.csv",
    csv_out="./expe_log/preds.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 155.12176537513733
Start prediction on validation dataset
Predictions done in 194.74140310287476
0.7091457082639498
0.70780087736029


Les prédictions de l'ensemble train ont un score de 0.701


Les prédictions de l'ensemble validation ont un score de 0.706

Prédictions sur les données reweighted avec 3 attributs sensibles (Age, Follow-up et View position)

In [4]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/metadata_rw1.csv",
    max_epochs = 2
)

GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:654: Checkpoint directory C:\Users\firda\Documents\L3\fairness IA\final\Fairness\expe_log exists and is not empty.
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\optim\lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params


Start training
Sanity Checking DataLoader 0:   0%|          | 0/2 [00:00<?, ?it/s]

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [03:21<00:00,  0.63it/s, v_num=1]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [03:22<00:00,  0.63it/s, v_num=1]
End of training 403.38070273399353


In [5]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/metadata_rw1.csv",
    csv_out="./expe_log/preds_rw1.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 142.75070571899414
Start prediction on validation dataset
Predictions done in 182.13253736495972
0.6927230869120843
0.6904444020598893


Les prédictions de l'ensemble train ont un score de 0.639


Les prédictions de l'ensemble validation ont un score de 0.620

Prédictions sur les données reweighted avec comme attribut sensible Age

In [7]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_age.csv",
    max_epochs = 2
)

GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:654: Checkpoint directory C:\Users\firda\Documents\L3\fairness IA\final\Fairness\expe_log exists and is not empty.
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\optim\lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params


Start training
Sanity Checking: |          | 0/? [00:00<?, ?it/s]

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [07:22<00:00,  0.29it/s, v_num=3]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [07:22<00:00,  0.29it/s, v_num=3]
End of training 837.2580513954163


In [8]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_age.csv",
    csv_out="./expe_log/preds_age.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 376.91709327697754
Start prediction on validation dataset
Predictions done in 474.5188159942627
0.688162214862288
0.6978828914743468


Les prédictions de l'ensemble train ont un score de 0.710


Les prédictions de l'ensemble validation ont un score de 0.714

Prédictions sur les données reweighted avec comme attribut sensible Follow-up

In [9]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_followup.csv",
    max_epochs = 2
)

GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:654: Checkpoint directory C:\Users\firda\Documents\L3\fairness IA\final\Fairness\expe_log exists and is not empty.
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\optim\lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params


Start training
Sanity Checking DataLoader 0:   0%|          | 0/2 [00:00<?, ?it/s]

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [05:09<00:00,  0.41it/s, v_num=4]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [05:09<00:00,  0.41it/s, v_num=4]
End of training 624.7406826019287


In [10]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_followup.csv",
    csv_out="./expe_log/preds_followup.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 391.2549457550049
Start prediction on validation dataset
Predictions done in 505.41283535957336
0.6212986119218645
0.6440968910928857


Les prédictions de l'ensemble train ont un score de 0.688


Les prédictions de l'ensemble validation ont un score de 0.695

Prédictions sur les données reweighted avec comme attribut sensible View Position

In [11]:
ckpt_path, ckpt_score = train_classifier(
    logdir="./expe_log/",
    datadir="./data/",
    csv="./data/weights_view.csv",
    max_epochs = 2
)

GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\callbacks\model_checkpoint.py:654: Checkpoint directory C:\Users\firda\Documents\L3\fairness IA\final\Fairness\expe_log exists and is not empty.
C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\torch\optim\lr_scheduler.py:62: UserWarning: The verbose parameter is deprecated. Please use get_last_lr() to access the learning rate.
  warnings.warn(

  | Name  | Type                  | Params | Mode 
--------------------------------------------------------
0 | model | ResNet                | 11.2 M | train
1 | bcm   | BinaryConfusionMatrix | 0      | train
--------------------------------------------------------
11.2 M    Trainable params


Start training
Sanity Checking:   0%|          | 0/2 [00:00<?, ?it/s]

C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


C:\Users\firda\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pytorch_lightning\trainer\connectors\data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.


Epoch 1: 100%|██████████| 128/128 [04:53<00:00,  0.44it/s, v_num=5]

`Trainer.fit` stopped: `max_epochs=2` reached.


Epoch 1: 100%|██████████| 128/128 [04:53<00:00,  0.44it/s, v_num=5]
End of training 616.7833323478699


In [12]:
pred_classifier(
    datadir="./data/",
    csv_in="./data/weights_view.csv",
    csv_out="./expe_log/preds_view.csv",
    ckpt_path = ckpt_path
)

Start prediction on train dataset
Predictions done in 148.30623364448547
Start prediction on validation dataset
Predictions done in 191.98968172073364
0.6934184573222811
0.7007438489414457


Les prédictions de l'ensemble train ont un score de 0.669


Les prédictions de l'ensemble validation ont un score de 0.680

Le reweighting apportant le meilleur résultats de prédictions est vle reweighting avec l'âge